Scripts/RemediationScripts/Remediate-SetSQLServerMinReqTLSVersion.ps1

```diff
@@ -0,0 +1,962 @@
+                            @{Expression={$_.Location};Label="Location";Width=7;Alignment="left"},
+                            @{Expression={$_.ServerVersion};Label="Server Version";Width=7;Alignment="left"},
+                            @{Expression={$_.minimalTlsVersion};Label="Minimal TLS Version";Width=7;Alignment="left"}
+
+        $sqlServersWithoutReqMinTLSVersion | Format-Table -Property $colsProperty -Wrap
+        Write-Host $([Constants]::SingleDashLine)
+    }
+
+    # Back up snapshots to `%LocalApplicationData%'.
+    $backupFolderPath = "$([Environment]::GetFolderPath('LocalApplicationData'))\AzTS\Remediation\Subscriptions\$($context.Subscription.SubscriptionId.replace('-','_'))\$($(Get-Date).ToString('yyyyMMddhhmm'))\SetSQLServerMinReqTLSVersion"
+
+    if (-not (Test-Path -Path $backupFolderPath))
+    {
+        New-Item -ItemType Directory -Path $backupFolderPath | Out-Null
+    }
+ 
+    Write-Host "`n[Step 4 of 5] Backing up SQL Server(s) details..."
+    Write-Host $([Constants]::SingleDashLine)
+    if ([String]::IsNullOrWhiteSpace($FilePath))
+    {        
+        if(-not $SkipBackup)
+        {
+            # Backing up SQL Server details.
+            $backupFile = "$($backupFolderPath)\sqlServersWithoutReqMinTLSVersion.csv"
+            $sqlServersWithoutReqMinTLSVersion | Export-CSV -Path $backupFile -NoTypeInformation
+            Write-Host "SQL Server(s) details have been successful backed up to $($backupFolderPath)" -ForegroundColor $([Constants]::MessageType.Update)
+            Write-Host $([Constants]::SingleDashLine)
+        }
+    }
+    else
+    {
+        Write-Host "Skipped as -FilePath is provided" -ForegroundColor $([Constants]::MessageType.Warning)
+        Write-Host $([Constants]::SingleDashLine)
+    }
+  
+    
+    if (-not $DryRun)
+    {  
+        # Here AutoRemediation switch is used as there is no need to take user input at BRS level if user has given consent to proceed with the remediation in AutoRemediation Script.
+        if(-not $AutoRemediation)
+        {
+
+            Write-Host "TLS Version will be set to required TLS version for all SQL Servers(s)." -ForegroundColor $([Constants]::MessageType.Warning)
+            Write-Host $([Constants]::SingleDashLine)
+            if (-not $Force)
+            {
+                Write-Host "Do you want to set TLS version to required TLS version for all SQL Server(s)? " -ForegroundColor $([Constants]::MessageType.Warning) -NoNewline
+                Write-Host $([Constants]::SingleDashLine)
+
+                $userInput = Read-Host -Prompt "(Y|N)" #TODO: 
+                Write-Host $([Constants]::SingleDashLine)
+                if($userInput -ne "Y")
+                {
+                    Write-Host "TLS version will not be changed for any SQL Server(s). Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+                    Write-Host $([Constants]::DoubleDashLine)
+                    break
+                }
```

Comment:
add another statement if user select 'Y'. something like TLS version will be set for the SQL servers...